### PR TITLE
[WIP] Warn that the "py" renderer can't be preceded by any other renderer

### PR DESCRIPTION
**Pull request by max-arnold**
### What does this PR do?

Add a warning that any renderer that comes before the "py" one is ignored.

### Previous Behavior

The `jinja` part in `jinja|py` renderer pipeline is silently ignored. This happens because the `py` renderer can only handle files:

https://github.com/saltstack/salt/blob/0ac7520c670a8d77b479355c38c19678a85a59ce/salt/renderers/py.py#L129
https://github.com/saltstack/salt/blob/0ac7520c670a8d77b479355c38c19678a85a59ce/salt/utils/templates.py#L535

### New Behavior

A warning has been added:

`The "py" renderer ignores previously processed results. The "#!jinja|py" pipeline would be equivalent to just "#!py"`

### Tests written?

Yes

### Commits signed with GPG?

No

### CODE REVIEW

File: salt/template.py
Comment (waynew) [inline salt/template.py]: Slices don't raise an IndexError, you can simplify this to just
```suggestion
        if 'py' in parts[1:]:
```

**Review by waynew — DISMISSED** [review]
Love it - just one suggestion to clean things up. Would it also make sense to add a `.. note::` to the renderer docs in the `py` section? Like, in https://docs.saltstack.com/en/latest/ref/renderers/all/salt.renderers.py.html#module-salt.renderers.py ?

**Comment by max-arnold** [comment]
@waynew Both suggestions were adressed

**Comment by max-arnold** [comment]
This is obviously a very minor usability fix (I've seen exactly two people who spent some time trying to understand why `jinja|py` doesn't work; one of them was me). However, there are several (mostly rhetorical) points to add:

1. The proper solution is to fix the `py` renderer, so it works the same way as any other renderer (i.e. can take a string as input)
2. There are no existing tests for the `py` renderer, so the entry barrier to fix it is higher (as per SEP10)
3. This PR contains good example of a test that is not an interface spec (it just documents the currently broken behavior)
4. The test could be safely removed to keep the test suite lean (i.e., the fix is an one-liner)

**Comment by max-arnold** [comment]
It looks like the https://github.com/saltstack/salt/pull/55390 makes my PR unnecessary!

File: tests/unit/test_template.py
Comment (waynew) [inline tests/unit/test_template.py:124]: Not sure, but would it be possible to dynamically grab the list of renderers from somewhere within Salt? I think we should at minimum show jinja/json (or, assert that there are at least one renderer in the list), but that would ensure that we're testing any potential future renderers.

If that's not possible with a reasonable amount of effort, then feel free to ignore the suggestion :slightly_smiling_face:

Comment (max-arnold) [reply tests/unit/test_template.py:124]: @waynew Can you help to improve this PR instead (it supersedes mine)? https://github.com/saltstack/salt/pull/55390

Comment (waynew) [reply tests/unit/test_template.py:124]: Should we go ahead and close this one then?

Comment (max-arnold) [reply tests/unit/test_template.py:124]: I’ll close mine once the other one is merged. That is why I set the title to WIP

**Comment by dwoz** [comment]
@waynew We good here now?

**Comment by waynew** [comment]
@dwoz I think that's entirely up to @max-arnold (and the other PR) - as he [mentioned](https://github.com/saltstack/salt/pull/55148#discussion_r408939919) there's another PR that actually makes it so py renderer *can* be be preceded by other renderers, so this warning won't be necessary.

So if that one gets done & merged, this one will no longer be necessary :)

**Pull request closed without merging**
